Source/NSDockTile.m

```diff
@@ -90,13 +108,112 @@ - (NSString *) badgeLabel
 
 - (void) setBadgeLabel: (NSString *)label
 {
+  NSImage *tempImage;
+  NSImageRep *rep;
+
   ASSIGNCOPY(_badgeLabel, label);
+
+  tempImage = [[NSImage alloc] initWithSize: [_appIconImage size]];
```

Comment:
This is now called whenever the label changes. Shouldn't the code rather go into a method like `setShowsApplicationBadge`?

Reply:
Now this code here makes even less sense.

Reply:
Did you see these comments here? The code here now belongs into the constructor or the `setShowsApplicationBadge`method in the case where we switch the display on and the off case would be to undo this.

Reply:
Yes, I saw then... but I also commented that it doesn't work. If I allocate the tempImage either in init or in setShowsApplicationBadge, then the application hangs and no badge is displayed.
Furthermore I commented that in all MacOS versions I have at hand, setShowsApplicationBadge does nothing beyond throwing a log of being not implemented.
This still means we can "respect" that setting of course, but I'd be careful in putting code there.